snuba/query/expressions.py

```diff
@@ -147,62 +165,92 @@ def visit_literal(self, exp: Literal) -> str:
             literal_str = f"date({exp.value.isoformat()})"
         else:
             literal_str = f"{exp.value}"
-        res = f"{self._get_line_prefix()}{literal_str}{self._get_alias_str(exp)}"
-        return res
+        return [f"{self._get_line_prefix()}{literal_str}{self._get_alias_str(exp)}"]
 
-    def visit_column(self, exp: Column) -> str:
+    def visit_column(self, exp: Column) -> List[str]:
         column_str = (
             f"{exp.table_name}.{exp.column_name}"
             if exp.table_name
             else f"{exp.column_name}"
         )
-        return f"{self._get_line_prefix()}{column_str}{self._get_alias_str(exp)}"
+        return [f"{self._get_line_prefix()}{column_str}{self._get_alias_str(exp)}"]
 
-    def visit_subscriptable_reference(self, exp: SubscriptableReference) -> str:
+    def visit_subscriptable_reference(self, exp: SubscriptableReference) -> List[str]:
         # we want to visit the literal node to format it properly
         # but for the subscritable reference we don't need it to
         # be indented or newlined. Hence we remove the prefix
         # from the string
-        literal_str = exp.key.accept(self)[len(self._get_line_prefix()) :]
+        literal_str = exp.key.accept(self)[0][len(self._get_line_prefix()) :]
+
+        # if the subscripted column is aliased, we wrap it with parens to make life
+        # easier for the viewer
+        column_str = (
+            f"({exp.column.accept(self)[0]})"
+            if exp.column.alias is not None
+            else f"{exp.column.accept(self)[0]}"
+        )
 
         # this line will already have the necessary prefix due to the visit_column
         # function
-        subscripted_column_str = f"{exp.column.accept(self)}[{literal_str}]"
+        subscripted_column_str = f"{column_str}[{literal_str}]"
         # after we know that, all we need to do as add the alias
-        return f"{subscripted_column_str}{self._get_alias_str(exp)}"
+        return [f"{subscripted_column_str}{self._get_alias_str(exp)}"]
 
-    def visit_function_call(self, exp: FunctionCall) -> str:
+    def visit_function_call(self, exp: FunctionCall) -> List[str]:
         self.__level += 1
-        # every param will have a newline in front of it so no need to put any space
-        # after the comma
-        param_str = ",".join([param.accept(self) for param in exp.parameters])
+        param_strs = [
+            f"{param},"
+            for param in chain(*(param.accept(self) for param in exp.parameters))
```

Comment:
```
param_strs = [
   f"{param.accept(self}," for param in exp.parameters
]
```
but I think your issue will be that you are turning all parameters to strings while they can be function calls themselves. Shouldn't they be added as lists to the current list ?